lib/Dialect/FIRRTL/FIRRTLIntrinsics.cpp

```diff
@@ -708,6 +710,239 @@ class CirctDPICallConverter : public IntrinsicConverter {
   }
 };
 
+//===----------------------------------------------------------------------===//
+// View intrinsic converter and helpers
+//===----------------------------------------------------------------------===//
+
+/// Implements the same behavior as DictionaryAttr::getAs<A> to return the
+/// value of a specific type associated with a key in a dictionary. However,
+/// this is specialized to print a useful error message, specific to custom
+/// attribute process, on failure.
+/// These attributes are created from input JSON.
+template <typename A>
+A tryGetAs(DictionaryAttr &dict, const Attribute &root, StringRef key,
+           Location loc, Twine path = Twine()) {
```

Comment:
Is there any way to unify this with the other `tryGetAs` utility or is it not worth it?

Reply:
I'll poke at this!

Reply:
Broke out the body for some reuse.